solc/CommandLineInterface.cpp

```diff
@@ -772,9 +773,7 @@ void CommandLineInterface::handleCombinedJSON()
 		}
 	}
 
-	string json = m_options.formatting.prettyJson ? jsonPrettyPrint(removeNullMembers(std::move(output))) :
-		jsonCompactPrint(removeNullMembers(std::move(output)));
-
+	string json = jsonPrint(removeNullMembers(std::move(output)), m_options.formatting.prettyJson);
```

Comment:
Could you add a command-line test for the new option?

These tests are in `test/cmdlineTests` and there are two kinds: Standard JSON tests (their names mostly start with `standard_`) and CLI test. The former contain `input.json` as input and the test script (`test/cmdlineTest.sh`) runs `solc --standard-json input.json` and ensures that the output matches `output.json`. The latter let you specify command-line options in `args` file and automatically add the `input.*` file if there's any in the test directory. To test this feature you need both `--standard-json` and ability to add more flags so you'll have to combine the two types.

Please take a look at how [test/cmdlineTests/standard_file_not_found](https://github.com/ethereum/solidity/tree/develop/test/cmdlineTests/standard_file_not_found) uses this flag. Then take [test/cmdlineTests/standard_default_success](https://github.com/ethereum/solidity/tree/develop/test/cmdlineTests/standard_default_success) and use its `input.json` in your test.

Reply:
I added a few tests but there is a problem with warning 3805 (pre-release compiler version) which gets removed from the output file by a sed command. Unfortunately, the pretty-printing option inserts line breaks into the JSON and the sed command works no longer. I added another sed command to delete the whole "errors" section on the condition that the "--pretty-json" option is present. It  not nice but it works.

Reply:
Well, this one works for tests added in this PR but I was hoping to be able to use the flag in other Standard JSON tests and there stripping all errors is a no-go. We need a better regex.

The command should remove only those `{}` blocks which contain a specific warning. We could narrow down your range but I'd have to dig in the doc to figure out if it's possible to make the `d` conditional but still affecting the whole content inside braces rather than just the lines that match. For now I only got this far:
```bash
sed -e '/error/,/]/ { /[{]/,/[}]/ { d } }'
```

Multi-line regexes in sed are generally not that great (though I have to say your range regex idea is pretty neat; I was not aware sed could do that) and the problem with other shell tools is often that they're not available on macOS (or are missing the flags GNU tools have). My suggestion would be to just use Python. We already depend on it anyway and I think we should generally rewrite most of the scripts in it - including `cmdlineTests.sh` but that's out of the scope of this PR.

So I think that in this PR we should just use this:
```bash
python3 - <<EOF
import re, sys
json = open("$stdout_path", "r").read()
json = re.sub(r"{[^{}]*Warning: This is a pre-release compiler version[^{}]*},?", "", json)
json = re.sub(r"},\s*]", "}]", json)
open("$stdout_path", "w").write(json)
EOF
```
Could be even shorter if we did not care about the comma that might remain after the last list element (`}, ]`) which is illegal in JSON.

Reply:
Two issues were revealed by CI tests:
 
1. The empty ``errors`` array needs to be also removed because in release builds there is no error 3508 at all and hence no ``errors`` array and the output does not match.
2. The regex does not remove the leading space before the error 3508. When the error itself gets removed, the leading space becomes trailing space making the ``scripts/check_style.sh`` script fail.

I added substitutions for both the issues although by removing whole ``errors`` the second problem also disappears. But the problem may manifest under different circumstances.

Reply:
I think this is fine. I'd just add some comments explaining what we're removing (added a review comment for that). Other than that it's great.